src/components/actionSheet/index.tsx

```diff
@@ -0,0 +1,265 @@
+
+  onOptionPress(optionIndex: number) {
+    _.invoke(this.props, `options[${optionIndex}].onPress`);
+    _.invoke(this.props, 'onDismiss');
+  }
+
+  handleRenderIcon = (option: ButtonProps) => {
+    // @ts-ignore
+    let source = option.icon;
+    if (!source) {
+      source = _.isFunction(option.iconSource) ? option.iconSource() : option.iconSource as ImageProps['source'];
+    }
+    return source && this.renderIcon(source);
+  };
+
+  renderIcon(iconSource: ImageSourcePropType) {
+    return <Image source={iconSource} resizeMode={'contain'} style={{width: 20, height: 20, marginRight: 16}}/>;
+  }
+
+  renderAction(option: ButtonProps, index: number) {
+    return (
+      <ListItem
+        style={{backgroundColor: 'transparent'}}
+        height={48}
+        key={index}
+        testID={option.testID}
+        onPress={() => this.onOptionPress(index)}
+        activeBackgroundColor={Colors.dark80}
+      >
+        <View row paddingL-16 flex centerV>
+          {this.handleRenderIcon(option)}
+          <Text text70 dark10 numberOfLines={1}>
+            {option.label}
+          </Text>
+        </View>
+      </ListItem>
+    );
+  }
+
+  renderActions() {
+    const {title, options, cancelButtonIndex, renderAction, optionsStyle} = this.props;
+    const optionsToRender = _.filter(options, (_option, index) => index !== cancelButtonIndex);
+
+    return (
+      <View style={[_.isEmpty(title) ? styles.listNoTitle : styles.listWithTitle, optionsStyle]}>
+        {_.isFunction(renderAction)
+          ? optionsToRender.map((option, index) => renderAction(option, index, this.onOptionPress))
+          : _.map(optionsToRender, this.renderAction)}
+      </View>
+    );
+  }
+
+  renderTitle() {
+    const {title} = this.props;
+
+    if (!_.isEmpty(title)) {
+      return (
+        <View height={56} paddingL-16 centerV>
+          <Text dark40 text70 style={{alignSelf: 'flex-start'}}>
+            {title}
+          </Text>
+        </View>
+      );
+    }
+  }
+
+  renderSheet() {
+    const {renderTitle} = this.props;
+    const {containerStyle} = this.props;
+    return (
+      <View style={[styles.sheet, containerStyle]}>
+        {_.isFunction(renderTitle) ? renderTitle() : this.renderTitle()}
+        {this.renderActions()}
+      </View>
+    );
+  }
+
+  render() {
+    const {useNativeIOS, visible, onDismiss, dialogStyle, onModalDismissed, testID, useSafeArea} =
+      this.props;
+
+    if (Constants.isIOS && useNativeIOS) {
+      return null;
+    }
+
+    return (
+      <Dialog
+        useSafeArea={useSafeArea}
+        testID={testID}
+        bottom
+        centerH
+        width="100%"
+        containerStyle={[styles.dialog, dialogStyle]}
+        visible={visible}
+        onDismiss={onDismiss}
+        onModalDismissed={onModalDismissed}
+        panDirection={PanningProvider.Directions.DOWN}
+      >
+        {this.renderSheet()}
+      </Dialog>
+    );
+  }
+}
+
+export default asBaseComponent(ActionSheet);
```

Comment:
This should be:
```suggestion
export default asBaseComponent<ActionSheetProps>(ActionSheet);
```
Once this is done you can run `npm run build:dev` to create the `.d.ts` file and (after you've added the change to the `generatedTypes/index`) you'll have proper autocomplete for it in `ActionSheetScreen`.